WordPress/Sniffs/WP/DeprecatedParameterValuesSniff.php

```diff
@@ -0,0 +1,196 @@
+	 *         )
+	 *     );
+	 */
+	protected $target_functions = array(
+		'add_settings_field' => array(
+			4 => array(
+				'misc' => array(
+					'alt'     => 'another settings group',
+					'version' => '3.0.0',
+				),
+				'privacy' => array(
+					'alt'     => 'another settings group',
+					'version' => '3.5.0',
+				),
+			),
+		),
+		'add_settings_section' => array(
+			4 => array(
+				'misc' => array(
+					'alt'     => 'another settings group',
+					'version' => '3.0.0',
+				),
+				'privacy' => array(
+					'alt'     => 'another settings group',
+					'version' => '3.5.0',
+				),
+			),
+		),
+		'bloginfo' => array(
+			1 => array(
+				'home' => array(
+					'alt'     => 'echo esc_url( home_url() )',
+					'version' => '2.2.0',
+				),
+				'siteurl' => array(
+					'alt'     => 'echo esc_url( home_url() )',
+					'version' => '2.2.0',
+				),
+				'text_direction' => array(
+					'alt'     => 'is_rtl()',
+					'version' => '2.2.0',
+				),
+			),
+		),
+		'get_bloginfo' => array(
+			1 => array(
+				'home' => array(
+					'alt'     => 'home_url()',
+					'version' => '2.2.0',
+				),
+				'siteurl' => array(
+					'alt'     => 'home_url()',
+					'version' => '2.2.0',
+				),
+				'text_direction' => array(
+					'alt'     => 'is_rtl()',
+					'version' => '2.2.0',
+				),
+			),
+		),
+		'register_setting' => array(
+			1 => array(
+				'misc' => array(
+					'alt'     => 'another settings group',
+					'version' => '3.0.0',
+				),
+				'privacy' => array(
+					'alt'     => 'another settings group',
+					'version' => '3.5.0',
+				),
+			),
+		),
+		'unregister_setting' => array(
+			1 => array(
+				'misc' => array(
+					'alt'     => 'another settings group',
+					'version' => '3.0.0',
+				),
+				'privacy' => array(
+					'alt'     => 'another settings group',
+					'version' => '3.5.0',
+				),
+			),
+		),
+	);
+
+	/**
+	 * Process the parameters of a matched function.
+	 *
+	 * @since 1.0.0
+	 *
+	 * @param int    $stackPtr        The position of the current token in the stack.
+	 * @param array  $group_name      The name of the group which was matched.
+	 * @param string $matched_content The token content (function name) which was matched.
+	 * @param array  $parameters      Array with information about the parameters.
+	 *
+	 * @return void
+	 */
+	public function process_parameters( $stackPtr, $group_name, $matched_content, $parameters ) {
+		$this->get_wp_version_from_cl();
+		$param_count = count( $parameters );
+		foreach ( $this->target_functions[ $matched_content ] as $position => $parameter_args ) {
+
+			// Stop if the position is higher then the total number of parameters.
+			if ( $position > $param_count ) {
+				break;
+			}
+
+			$parameter_position = $this->phpcsFile->findNext(
+				Tokens::$emptyTokens,
+				$parameters[ $position ]['start'],
+				$parameters[ $position ]['end'] + 1,
+				true
+			);
+
+			$matched_parameter = $this->strip_quotes( $this->tokens[ $parameter_position ]['content'] );
```

Comment:
This test case is not valid PHP. I tried a few different things but I don't think `$parameter_position` will ever return false as checks earlier make sure of that.

I can add check before if you wish as defensive code.

Reply:
The above example will be valid code as of PHP 7.3 (trailing comma's will be allowed in function calls). Even so, while I agree that the code sample was not valid before that, the sniff should function without throwing notices, even when parse/fatal errors are encountered and it would not at this moment.
So yes, this simple defensive coding check is warranted.

Reply:
I will add the check but before I do I wanted to mention that `add_settings_field( '', '', '', /* deliberately not passed */ );` does not cause any errors with the current code.

Reply:
> does not cause any errors with the current code.

... but it did look at token 0 for the content - which in this case is the PHP open tag. (false is being cast to 0)